src/languages/swift.js

```diff
@@ -283,6 +288,143 @@ export default function(hljs) {
+    keywords: "_",
+    relevance: 0
+  };
+  // Matches tuples as well as the parameter list of a function type.
+  const TUPLE = {
+    begin: /\(/,
+    end: /\)/,
+    relevance: 0,
+    keywords: KEYWORDS,
+    contains: [
+      'self',
+      TUPLE_ELEMENT_NAME,
+      ...COMMENTS,
+      ...KEYWORD_MODES,
+      ...BUILT_INS,
+      ...OPERATORS,
+      NUMBER,
+      STRING,
+      ...IDENTIFIERS,
+      ...ATTRIBUTES,
+      TYPE
+    ]
+  };
+
+  // https://docs.swift.org/swift-book/ReferenceManual/Declarations.html#ID362
+  // Matches both the keyword func and the function title.
+  // Grouping these lets us differentiate between the operator function <
+  // and the start of the generic parameter clause (also <).
+  const FUNCTION_TITLE = {
+    begin: lookahead(/\bfunc\b/),
+    relevance: 0,
+    contains: [
+      {
+        className: 'keyword',
+        begin: /\bfunc\b/
+      },
+      {
+        begin: /\s+/,
+        relevance: 0
+      },
+      {
+        className: 'title',
+        begin: either(QUOTED_IDENTIFIER.begin, Swift.identifier, Swift.operator),
+        // Required to make sure the opening < of the generic parameter clause
+        // isn't parsed as a second title.
+        endsParent: true,
+        relevance: 0
+      }
+    ]
+  };
+  const GENERIC_PARAMETERS = {
+    begin: /\s*</,
+    end: />/,
+    contains: [
+      ...COMMENTS,
+      TYPE
+    ]
+  };
+  const FUNCTION_PARAMETER_NAME = {
+    begin: either(
+      lookahead(concat(Swift.identifier, /\s*:/)),
+      lookahead(concat(Swift.identifier, /\s+/, Swift.identifier, /\s*:/))
+    ),
+    end: lookahead(/:/),
+    relevance: 0,
+    contains: [
+      {
+        className: 'keyword',
+        begin: /\b_\b/
+      },
+      {
+        className: 'params',
+        begin: Swift.identifier
+      }
+    ]
+  };
+  const FUNCTION_PARAMETERS = {
+    begin: /\s*\(/,
+    end: /\)/,
+    keywords: KEYWORDS,
+    contains: [
+      FUNCTION_PARAMETER_NAME,
+      ...COMMENTS,
+      ...KEYWORD_MODES,
+      ...OPERATORS,
+      NUMBER,
+      STRING,
+      ...ATTRIBUTES,
+      TYPE,
+      TUPLE
+    ],
+    illegal: /["']/
+  };
+  const FUNCTION = {
+    className: 'function',
+    begin: lookahead(/\bfunc\b/),
+    contains: [
+      FUNCTION_TITLE,
+      GENERIC_PARAMETERS,
+      FUNCTION_PARAMETERS
+    ],
+    illegal: /\[|%/
+  };
+
+  // https://docs.swift.org/swift-book/ReferenceManual/Declarations.html#ID375
+  // https://docs.swift.org/swift-book/ReferenceManual/Declarations.html#ID379
+  const INIT_SUBSCRIPT_TITLE = {
+    begin: lookahead(/\b(subscript|init[?!]?)\s*[<(]/),
+    end: lookahead(/[<(]/),
+    relevance: 0,
+    contains: [
+      {
+        className: 'keyword',
+        begin: /subscript|init[?!]?/
+      },
+      {
+        begin: /\s+/,
+        relevance: 0
+      }
+    ]
+  };
+  const INIT_SUBSCRIPT = {
+    className: 'function',
+    begin: lookahead(/\b(subscript|init[?!]?)\s*[<(]/),
```

Comment:
```suggestion
  const INIT_SUBSCRIPT = {
    className: 'function',
    begin: /\b(subscript|init[?!]?)\s*[<(]/,
    keywords: { // ... 
```

Reply:
You will need $pattern here of course to deal with the `?!`...

Reply:
You should really look at `beginKeywords` also as that could also be used in MANY of these cases... and is even simpler than `keywords` since it uses avoids the need for $pattern for more complex cases.

Reply:
Does `beginKeyword` support matching keywords like `init!`? Or is it tied to the default $pattern?

Reply:
It's just sugar around `either`:

```
      if (mode.beginKeywords) {
        mode.begin = '\\b(' + mode.beginKeywords.split(' ').join('|') + ')(?!\\.)(?=\\b|\\s)';
```

Reply:
Oh wait, yeah sorry you might still need to adjust `$pattern`... beginKeywords uses an either for the match but then uses the keywords engine for the highlighting.

Reply:
This:

```
    begin: lookahead(/\b(subscript|init[?!]?)\s*[<(]/),
    contains: [
      {
        className: 'keyword',
        begin: /subscript|init[?!]?/
      },
```

Becomes (off the top of my head):

```js
    begin: /\b(subscript|init[?!]?)\s*[<(]/,
    keywords: { 
      keyword: "subscript init init? init!",
      $pattern: /\w+[?!]?/
```

And now there is no potential sequencing randomness doe to `contains` since everything is grouped in a single cluster.

Reply:
Fixed!

Reply:
This is how simple function would be if not for the `<` craziness. :)